### PR TITLE
[Merged by Bors] - refactor(AlgebraicGeometry): replace `OpenCover` by `Cover @IsOpenImmersion`

### PR DESCRIPTION
**Pull request by chrisflav**
This prepares for more general notions of `IsLocalAtTarget` etc. notions and is one option for defining sites defined by morphism properties.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Cover/MorphismProperty.lean [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    · exact 𝒰.map_prop _
+
+/-- A morphism property of schemes is said to preserve joint surjectivity, if
+for any pair of morphisms `f : X ⟶ S` and `g : Y ⟶ S` where `g` satisfies `P`,
+any pair of points `x : X` and `y : Y` with `f x = g y` can be lifted to a point
+of `X ×[S] Y`.
+
+In later files, this will be automatic, since this holds for any morphism `g`. But at
+this early stage in the import tree, we only know it for open immersions. -/
+class IsJointlySurjectivePreserving (P : MorphismProperty Scheme.{u}) where
```
Comment (erdOne) [inline Mathlib/AlgebraicGeometry/Cover/MorphismProperty.lean:157]: Is this instance provided for all morphism properties?

Comment (chrisflav) [reply Mathlib/AlgebraicGeometry/Cover/MorphismProperty.lean:157]: I forgot that part, thanks! Fixed.

Comment (erdOne) [reply Mathlib/AlgebraicGeometry/Cover/MorphismProperty.lean:157]: Can you point to this file in the docstring of `IsJointlySurjectivePreserving`?

Comment (chrisflav) [reply Mathlib/AlgebraicGeometry/Cover/MorphismProperty.lean:158]: Done

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**